core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -294,21 +293,57 @@ private GraphHopper setUnsafeMemory()
     }
 
     /**
+     * Wrapper method for {@link GraphHopper#setCHWeighting(List)}
+     */
+    public GraphHopper setCHWeighting( String... weightingNames )
+    {
+        return this.setCHWeighting(Arrays.asList(weightingNames));
+    }
+
+    /**
      * Enables the use of contraction hierarchies to reduce query times. Enabled by default.
      * <p>
      *
-     * @param weighting can be "fastest", "shortest" or your own weight-calculation type.
+     * @param weightingList A list containing multiple weightings like:
+     *                     "fastest", "shortest" or your own weight-calculation type.
      */
-    public GraphHopper setCHWeighting( String weighting )
+    public GraphHopper setCHWeighting( List<String> weightingList )
     {
         ensureNotLoaded();
-        chWeightingStr = weighting.toLowerCase();
+
+        if (weightingList.isEmpty())
+            throw new IllegalArgumentException("It is not allowed to pass an emtpy weightingList");
+
+        this.chWeightingList.clear();
+        for (String weight : weightingList)
+        {
+            weight = weight.toLowerCase();
+            weight = weight.trim();
+            this.chWeightingList.add(weight);
+        }
         return this;
     }
 
+    public List<String> getAllCHWeightings()
```

Comment:
> After writing the above I just realized, that getWeightingForCH and getCHWeighting is probably irritating.

Hmmh, that is right. I would get rid of it and just keep `getCHWeightings` and `getWeightingForCH`. Currently new GH versions broke the Java API and if it is properly documented in the changelog.txt there was no problem or even questions in the forum.

BTW: Maybe we should rename `getWeightingForCH` to `findCHWeighting` to indicate the small overhead? But would be a different issue